src/hyperlight_host/src/hypervisor/kvm.rs

```diff
@@ -438,6 +439,21 @@ impl Hypervisor for KVMDriver {
             dbg_mem_access_fn,
         )?;
 
+        // The guest may have chosen a different stack region. If so, we drop usage of our tmp stack.
+        let mut hyperlight_peb = self.mem_sections.read_hyperlight_peb()?;
```

Comment:
This circumvents our abtraction for shared memory (applies to all drivers). We should re-think this